ocaml/asmcomp/cmm_helpers.ml

```diff
@@ -1356,9 +1356,14 @@ let default_prim name =
 let int64_native_prim name arity ~alloc =
   let u64 = Primitive.Unboxed_integer Primitive.Pint64 in
   let rec make_args = function 0 -> [] | n -> u64 :: make_args (n - 1) in
+  let effects, coeffects =
+    if alloc
+    then Primitive.Arbitrary_effects, Primitive.No_coeffects
```

Comment:
This should be `Only_generative_effects` not `Arbitrary_effects`, I think.

Reply:
I'm not sure `Only_generative_effects` is appropriate. Despite the `alloc` name, what this really means is that the C primitive might raise (it's used for `div` and `mod`). I would assume that `Arbitrary_effects` better represents this, though it's not very satisfying either.

Reply:
The reason I set it to Arbitrary_effects is exactly what @lthls  said. I am not sure it is safe to set it as No_coeffects under this meaning.

Reply:
Ah, sorry -- this is `alloc` in the sense of the `noalloc` attribute, right?  In which case it should presumably be arbitrary effects and arbitrary coeffects?

Reply:
This PR only wants to update `semantics_of_primitives.ml`  in middle_end, and it is based on #18. The changes you commented on are from #18, sorry about the confusion.